### PR TITLE
EVAKA-4006 e2e tests for mobile pairing flow

### PR DESCRIPTION
**Pull request by massproduction**
#### Summary
Added e2e tests for mobile pairing flow.

These tests only test only side of the pairing flow at a time and "mock" the other side with dev-api calls.

### CODE REVIEW

File: frontend/e2e-test/test/e2e/specs/5_employee/unit-acl.spec.ts [hunk 1]
```diff
@@ -72,6 +79,8 @@ fixture('Employee - Unit ACL')
   })
   .afterEach(logConsoleMessages)
   .after(async () => {
+    if (pairingId) await deletePairing(pairingId)
+    if (deviceId) await deleteMobileDevice(deviceId)
```
Comment (Joosakur) [inline frontend/e2e-test/test/e2e/specs/5_employee/unit-acl.spec.ts:83]: set these back to null after delete?

File: service/src/main/kotlin/fi/espoo/evaka/shared/dev/DevApi.kt [hunk 2]
```diff
@@ -609,6 +618,64 @@ RETURNING id
...
+            val pairing = tx.initPairing(body.unitId)
+
+            asyncJobRunner.plan(
+                tx = tx,
+                payloads = listOf(GarbageCollectPairing(pairingId = pairing.id)),
+                runAt = ZonedDateTime
+                    .ofInstant(pairing.expires, zoneId)
+                    .plusDays(1)
+                    .toInstant()
+            )
```
Comment (Joosakur) [inline service/src/main/kotlin/fi/espoo/evaka/shared/dev/DevApi.kt:662]: not needed in dev-api imo

File: service/src/main/kotlin/fi/espoo/evaka/shared/dev/DevApi.kt [hunk 3]
```diff
@@ -647,6 +714,14 @@ fun Handle.clearDatabase() = listOf(
     execute("DELETE FROM $it")
 }
 
+fun Handle.deletePairing(id: UUID) {
+    execute("DELETE FROM pairing WHERE id = ?", id)
+}
+
+fun Handle.deleteMobileDevice(id: UUID) {
```
Comment (Joosakur) [inline service/src/main/kotlin/fi/espoo/evaka/shared/dev/DevApi.kt:721]: replace deprecated `fun Handle.foo` with `fun Database.Transaction.foo`

File: frontend/e2e-test/test/e2e/dev-api/index.ts [hunk 4]
```diff
@@ -580,3 +580,91 @@ export async function cleanUpApplicationEmails(): Promise<void> {
...
+  expires: Date
+  status: PairingStatus
+  mobileDeviceId: UUID | null
+}
+
+export async function postPairingChallenge(
+  challengeKey: string
+): Promise<PairingResponse> {
+  try {
+    const { data } = await devClient.post<PairingResponse>(
```
Comment (Joosakur) [inline frontend/e2e-test/test/e2e/dev-api/index.ts:604]: This is also fine, but I'm wondering if we could do these things through standard API without duplicating these to dev-api. Maybe @Gekkio can comment?

File: frontend/e2e-test/test/e2e/specs/6_mobile/pairing.spec.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  const responseKey = await pairingFlow.responseKey.textContent
+  const pairingResponse = await postPairingResponse(
+    res.id,
+    res.challengeKey,
+    responseKey
+  )
+  pairingId = pairingResponse.id
+  deviceId = pairingResponse.mobileDeviceId
+
+  await t.expect(pairingFlow.mobilePairingTitle3.exists).ok()
```
Comment (Joosakur) [inline frontend/e2e-test/test/e2e/specs/6_mobile/pairing.spec.ts:101]: I think we should verify that the session has been created correctly, i.e. go to the mobile app and verify that it opens the correct unit and is able to fetch some data.

**Review by Joosakur — CHANGES_REQUESTED** [review]

File: frontend/e2e-test/test/e2e/specs/6_mobile/pairing.spec.ts (hunk 5 above)
Comment (massproduction) [reply frontend/e2e-test/test/e2e/specs/6_mobile/pairing.spec.ts:107]: Added check that in the group selector (the next page) we get the unit's name.

**Review by Joosakur — APPROVED** [review]

**Pull request merged**